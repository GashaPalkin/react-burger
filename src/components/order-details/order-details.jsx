import React from "react";
import PropTypes from "prop-types";
import imageDone from "../../images/done.png";

export const OrderDetails = ({ orderNumber  }) => {
  return (
    <React.Fragment>
        <p className="text text_type_digits-large mt-8">
           {orderNumber}
          </p>
          <p className="text text_type_main-medium mt-8">
            идентификатор заказа
          </p>
          <img
            className="m-15"
            src={imageDone}
            alt="done"
            height={120}
            width={120}
          />
          <p className="text text_type_main-default">
            Ваш заказ начали готовить
          </p>
          <p className="text text_type_main-default text_color_inactive mt-2 mb-10">
            Дождитесь готовности на орбитальной станции
          </p>
    </React.Fragment>
  );
};

//Типизация компонентов
OrderDetails.propTypes = {
  orderNumber: PropTypes.number
}
