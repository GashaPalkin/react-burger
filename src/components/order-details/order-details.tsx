import { FC } from "react";
import imageDone from "../../images/done.png";

interface OrderDetailProps {
  orderNumber: number;
}

export const OrderDetails: FC<OrderDetailProps> = ({ orderNumber }) => {
  return (
    <>
      <p className="text text_type_digits-large mt-8">{orderNumber}</p>
      <p className="text text_type_main-medium mt-8">идентификатор заказа</p>
      <img
        className="m-15"
        src={imageDone}
        alt="done"
        height={120}
        width={120}
      />
      <p className="text text_type_main-default">Ваш заказ начали готовить</p>
      <p className="text text_type_main-default text_color_inactive mt-2 mb-10">
        Дождитесь готовности на орбитальной станции
      </p>
    </>
  );
};
