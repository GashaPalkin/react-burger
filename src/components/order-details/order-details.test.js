import renderer from 'react-test-renderer';
import { OrderDetails } from './order-details';

describe('Тест компонента OrderDetails', () => {
   it('Ордер рендерится без ошибок', () => {
      const tree = renderer
        .create(<OrderDetails orderNumber={112233} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    }); 
})
