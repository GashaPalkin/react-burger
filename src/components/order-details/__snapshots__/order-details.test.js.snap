// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента OrderDetails Ордер рендерится без ошибок 1`] = `
Array [
  <p
    className="text text_type_digits-large mt-8"
    data-testid="orderNumber"
  >
    112233
  </p>,
  <p
    className="text text_type_main-medium mt-8"
  >
    идентификатор заказа
  </p>,
  <img
    alt="done"
    className="m-15"
    height={120}
    src="done.png"
    width={120}
  />,
  <p
    className="text text_type_main-default"
  >
    Ваш заказ начали готовить
  </p>,
  <p
    className="text text_type_main-default text_color_inactive mt-2 mb-10"
  >
    Дождитесь готовности на орбитальной станции
  </p>,
]
`;
